### ISSUE TITLE
A11y Dropdown keyboard issue

### ISSUE THREAD

**Issue by mlmoravek** [oruga-ui/oruga#1115]
The Dropdown component can't be controlled using the keyboard.

According to https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/ users should be able to use the arrow keys to highlight items and scroll through the list, and select items using Enter or the spacebar. I believe this would require tracking the currently highlighted item using state.

A quick and "dirty" but simple solution could be to register a `@keypress` handler on the dropdown item like this:

```ts
function onKeyPress(event: KeyboardEvent): void {
    if (!isClickable.value) return;
    if (event.key === "Enter" || event.key === " ") {
        event.preventDefault();
        parent.value.selectItem(itemValue as T);
        emits("click", itemValue as T, event);
    }
}
```

This would at least allow using the dropdown using keyboard only, since it is currently possible to tab through the items (not following spec either, but well).

_Originally posted by @jh0ker in https://github.com/oruga-ui/oruga/issues/596#issuecomment-2474097370_
